lib/views/pages/home_page.dart

```diff
@@ -0,0 +1,60 @@
+import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/utils/ui/circular_value.dart";
+import "package:proxima/viewmodels/login_view_model.dart";
+import "package:proxima/viewmodels/profile_view_model.dart";
+import "package:proxima/views/navigation/routes.dart";
+
+class HomePage extends HookConsumerWidget {
+  static const logoutButtonKey = Key("logout");
+
+  const HomePage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Home page :)"),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const UserProfile(),
+            const SizedBox(
+              height: 20,
+            ),
+            ElevatedButton(
+              key: logoutButtonKey,
+              onPressed: () {
+                ref.read(loginServiceProvider).signOut();
+
+                // Go to login page when the user is logged out
+                Navigator.pushNamedAndRemoveUntil(
+                  context,
+                  Routes.login.name,
+                  (route) => false,
+                );
+              },
+              child: const Text("Logout"),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class UserProfile extends HookConsumerWidget {
+  const UserProfile({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return CircularValue(
```

Comment:
I am not sure to understand why we need the `CircularValue` here.

For me, we are displaying the login page (`Navigator.pushReplacementNamed`) under the condition that we received the `userProvider`. So we will never see the `CircularValue`.

The branch is targeted at the authentication system part, but I am just making sure that I have understood everything 😄 .

Reply:
Linked to that, we also have the `ProfileViewModel ` as `AsyncNotifier<ProfileData>` for the same reason(`AsyncNotifier<ProfileData>` is for the `CircularValue` input parameter).

If I am missing something, can you educate me please 😄

Reply:
If I may add, I don't see any circle loading between the moment I click the log in button and the moment I get to this page on my machine.

Reply:
Yes so this is an interesting question, I anticipated a network request that will happen in the `ProfileViewModel` (that does not exist today but will certainly in the future).

The `CircularValue` is a helper widget that displays a small loading circle while waiting for the data to arrive.

This should explain why you don't see the loading yet (as there is no outgoing request in the user profile view model).  
If you want to try it for yourself, you can add a `await Future.delayed(const Duration(seconds: 5));` before returning from the `build` method of the `ProfileViewModel` to mimic the behavior of querying the database. You should see a loading indicator instead of the welcoming text for 5 seconds.

Hope it clarifies things.